### PR TITLE
feat: Enable expert mode

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**
feat: Enable expert mode (#16034)

Display technical group IDs, attribute IDs and attribute domain values, KB information in the header section of the interactive configuration page

Closes CXSPA-857

### CODE REVIEW

**Comment by Uli-Tiger** [comment]
@Larisa-Staroverova:
- Can you format the PR title so it can be used as commit message header.
- Can you please write a PR description, that could be used as merge commit message body.
Guidelines: https://sap.github.io/spartacus-docs/commit-guidelines/#commit-message-guidelines

File: feature-libs/product-configurator/common/components/service/configurator-router-extractor.service.spec.ts [hunk 1]
```diff
@@ -181,6 +181,18 @@ describe('ConfigRouterExtractorService', () => {
...
+      let routerData: ConfiguratorRouter.Data;
+      serviceUnderTest
+        .extractRouterData()
+        .subscribe((data) => {
+          routerData = data;
+          expect(routerData.expMode).toBe(true);
+        })
+        .unsubscribe();
+    });
   });
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/service/configurator-router-extractor.service.spec.ts:221]: can you plz add at least one test for the default behaviour, so 
when queryParam expMode is not present
then expect(routerData.expMode).toBe(false);

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.html [hunk 2]
```diff
@@ -1,3 +1,7 @@
+<div *ngIf="!attribute.visible" class="cx-hidden-msg">
+  <cx-icon [type]="iconTypes.WARNING" aria-hidden="true"></cx-icon>
+  {{ 'configurator.attribute.notVisibleAttributeMsg' | cxTranslate }}
+</div>
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.html]: shouldn't this only be displayed in case expMode=true?

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.html [hunk 3]
```diff
@@ -35,7 +39,7 @@
   <ng-container *ngIf="isAttributeGroup(); else conflictGroup">
     <ng-container *ngIf="isNavigationToConflictEnabled(); else withoutLink">
       <a
-        class="cx-conflict-msg link cx-action-link"
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.html:38]: why did you remove 'cx-conflict-msg' class?

File: feature-libs/product-configurator/rulebased/components/conflict-description/configurator-conflict-description.component.html [hunk 4]
```diff
@@ -1,4 +1,7 @@
 <ng-container *ngIf="displayConflictDescription(currentGroup)">
   <cx-icon [type]="iconTypes.WARNING" aria-hidden="true"></cx-icon>
-  {{ currentGroup.name }}
+  <div>
+    <span>{{ currentGroup.description }}</span>
+    <span>{{ currentGroup.name }}</span>
+  </div>
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/conflict-description/configurator-conflict-description.component.html]: how does this relate to expert mode?

**Review by Uli-Tiger — COMMENTED** [review]
First Review Batch - will continue later

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts [hunk 5]
```diff
@@ -76,6 +78,9 @@ export class ConfiguratorFormComponent implements OnInit {
               }
             });
         }
+        if (routingData.expMode) {
+          this.configExpertModeService.setExpMode(routingData.expMode);
+        }
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts]: seem like that there was no unit test added for this condition

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts [hunk 6]
```diff
@@ -100,4 +105,8 @@ export class ConfiguratorFormComponent implements OnInit {
   createGroupId(groupId?: string): string | undefined {
     return this.configUtils.createGroupId(groupId);
   }
+
+  get expMode(): Observable<boolean> {
+    return this.configExpertModeService.getExpMode();
+  }
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts]: seem like that there was no unit test added for this condition

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts [hunk 7]
```diff
@@ -1515,4 +1522,51 @@ describe('ConfigurationGroupMenuComponent', () => {
...
+      const configForExpMode =
+        ConfigurationTestData.productConfigurationWithConflicts;
+      initialize();
+
+      expect(component.getGroupMenuTitle(configForExpMode.groups[0])).toEqual(
+        configForExpMode.groups[0].description
+      );
+    });
+
+    it('should return conflict group title for expert mode', () => {
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts]:
```suggestion
it('should return only conflict group description as title even if expert mode is on', () => {
```

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts [hunk 8]
```diff
@@ -1515,4 +1522,51 @@ describe('ConfigurationGroupMenuComponent', () => {
...
+        mockProductConfiguration.groups[0].description +
+        ' / [' +
+        mockProductConfiguration.groups[0].name +
+        ']';
+      expect(
+        component.getGroupMenuTitle(mockProductConfiguration.groups[0])
+      ).toEqual(groupMenuTitle);
+    });
+
+    it('should return conflict header group title for expert mode', () => {
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts]:
```suggestion
    it('should return only conflict header group description as title even if expert mode is on', () => {
```

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts [hunk 9]
```diff
@@ -1515,4 +1522,51 @@ describe('ConfigurationGroupMenuComponent', () => {
...
+    it('should return group title', () => {
+      spyOn(configExpertModeService, 'getExpMode').and.returnValue(of(false));
+      initialize();
+
+      expect(
+        component.getGroupMenuTitle(mockProductConfiguration.groups[0])
+      ).toEqual(mockProductConfiguration.groups[0].description);
+    });
+
+    it('should return group title for expert mode', () => {
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts]:
```suggestion
    it('should return group description and name as title when expert mode is on', () => {
```

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts [hunk 10]
```diff
@@ -1515,4 +1522,51 @@ describe('ConfigurationGroupMenuComponent', () => {
       );
     });
   });
+
+  describe('getGroupMenuTitle', () => {
+    it('should return group title', () => {
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts]:
```suggestion
    it('should return only group description as title when expert mode if off', () => {
```

**Comment by Uli-Tiger** [comment]
some changed constructors were marked with
//TODO(CXSPA-1014): take care about constructor deprecation
some not. can you please add the TODO to all relevant changes?

File: feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.ts [hunk 11]
```diff
@@ -17,12 +24,17 @@ import { ConfiguratorCommonsService } from '../../core/facade/configurator-commo
...
+  configuration$: Observable<Configurator.Configuration> =
+    this.configRouterExtractorService.extractRouterData().pipe(
+      switchMap((routerData) => {
+        return this.configuratorCommonsService.getConfiguration(
+          routerData.owner
+        );
+      })
+    );
+
+  product$: Observable<Product | undefined> = this.configuration$
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.ts:42]: no unit test added or adapted?

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts [hunk 12]
```diff
@@ -399,6 +407,27 @@ describe('OccConfiguratorVariantNormalizer', () => {
...
+      expect(result.kbKey.kbVersion).toEqual(configuration.kbKey.kbVersion);
+      expect(result.kbKey.kbBuildNumber).toEqual(
+        configuration.kbKey.kbBuildNumber
+      );
+    });
+
+    it('should convert a configuration with undefined kb key', () => {
+      configuration.kbKey = {};
+      const result = occConfiguratorVariantNormalizer.convert(configuration);
+      expect(result.kbKey).toBeDefined();
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: does it really make sense to add a kbkey object with undefined fields to the configuration? - i would expect that kbkey itself is undefined, if no kbkey is provided.

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts [hunk 13]
```diff
@@ -34,6 +34,12 @@ export class OccConfiguratorVariantNormalizer
       productCode: source.rootProduct,
       groups: [],
       flatGroups: [],
+      kbKey: {
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts]: only initialize kbkey, if source.kbkey is defined?

**Review by Uli-Tiger — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.html (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.html]: Normally only visible attributes come, thus `This attribute is not visible to the user` message will not be displayed.
There is no need to check additionally whether the expert mode is set

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.html (hunk 3 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.html:38]: See Slack

File: feature-libs/product-configurator/rulebased/components/conflict-description/configurator-conflict-description.component.html (hunk 4 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/conflict-description/configurator-conflict-description.component.html]: You are right. 
I reverted the changes.
PS: it was relevant for the discussion with Anthony

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts (hunk 5 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts (hunk 12 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: I extended configuration with corresponding kbKex data: 
image
That's why I set configuration.kbKey to empty object

Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: What i meant is:
your test expects that, if the kbKey is undefined in the OCC configuration, that it is still defined in the model after converting (line 425), but with all fields being undefined (line 425-430). But why - do we have any specific use case for such a behaviour?
I would expect, if the kbKey is undefined in the occ configuration, that it is also undefined in the model, after converting. In the model the kbKey is marked as optional, so it is legit to set it to undefined.

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts (hunk 13 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.ts]: Agree. Done!

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts (hunk 12 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts (hunk 6 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts]: Covered indirectly.

File: feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.ts (hunk 11 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.ts:42]: Done

File: feature-libs/product-configurator/common/components/service/configurator-router-extractor.service.spec.ts (hunk 1 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/common/components/service/configurator-router-extractor.service.spec.ts:221]: @Larisa-Staroverova: did you saw this?

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts (hunk 6 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts]: Covered indirectly does not mean tested, or?
if we would always emit true or false - would a test fail?

**Comment by Uli-Tiger** [comment]
@Larisa-Staroverova can plz. check the remaining two open conversations - thx!

**Comment by Uli-Tiger** [comment]
@Larisa-Staroverova PS: Could you plz. list all changed constructors in [CXSPA-1014](https://jira.tools.sap/browse/CXSPA-1014)

**Comment by Larisa-Staroverova** [comment]
> @Larisa-Staroverova PS: Could you plz. list all changed constructors in [CXSPA-1014](https://jira.tools.sap/browse/CXSPA-1014)

Done!

**Comment by Larisa-Staroverova** [comment]
> some changed constructors were marked with //TODO(CXSPA-1014): take care about constructor deprecation some not. can you please add the TODO to all relevant changes?

I followed this [SPA documentation](https://sap.github.io/spartacus-docs/breaking-changes/#adding-new-constructor-dependencies-in-minor-versions)

File: feature-libs/product-configurator/common/components/service/configurator-router-extractor.service.spec.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/common/components/service/configurator-router-extractor.service.spec.ts:221]: Done!
PS: this comment is not shown in this [view](https://github.com/SAP/spartacus/pull/16034/files)

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts (hunk 6 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/form/configurator-form.component.ts]: Done!

**Review by Uli-Tiger — DISMISSED** [review]

File: feature-libs/product-configurator/rulebased/core/services/configurator-expert-mode.service.ts [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { Observable, ReplaySubject } from 'rxjs';
+
+/**
+ * This implementation is OCC specific.
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/core/services/configurator-expert-mode.service.ts:11]: This is misleading, as the service itself is not OCC specific.
Maybe we could state that CPQ does not provide an expert mode

File: feature-libs/product-configurator/rulebased/root/events/configurator-logout-event.listener.ts [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+
+import { Injectable, OnDestroy } from '@angular/core';
+import { EventService, LogoutEvent } from '@spartacus/core';
+import { merge, Subscription } from 'rxjs';
+import { ConfiguratorExpertModeService } from '../../core/services/configurator-expert-mode.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ConfiguratorLogoutEventListener implements OnDestroy {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/root/events/configurator-logout-event.listener.ts]: After core team confirms that event listener can be lazily loaded, and following @schuppel 's suggestion:
Let's have the event listener in the core entry point in feature-libs/product-configurator/rulebased/core/event.
Please make sure to export it to the public API of the core endpoint.
Create a dedicated module for the event listener and import this module in RulebasedConfiguratorCoreModule.
Create new action that removes all product bound configurations, the corresponding effect here /rulebased/core/state/effects/configurator-basic.effect.ts

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts [hunk 16]
```diff
@@ -1515,4 +1521,59 @@ describe('ConfigurationGroupMenuComponent', () => {
       );
     });
   });
+
+  describe('getGroupMenuTitle', () => {
+    it('should return only group description as title when expert mode if off', () => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.spec.ts]:
```suggestion
    it('should return only group description as title when expert mode is off', () => {
```

File: feature-libs/product-configurator/rulebased/components/attribute/types/read-only/configurator-attribute-read-only.component.html [hunk 17]
```diff
@@ -1,4 +1,7 @@
-<div id="{{ createAttributeIdForConfigurator(attribute) }}">
+<div
+  id="{{ createAttributeIdForConfigurator(attribute) }}"
+  *ngIf="attribute.values !== undefined && attribute.values.length > 0"
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/read-only/configurator-attribute-read-only.component.html:3]: Why was that needed for expert mode?

File: feature-libs/product-configurator/rulebased/components/attribute/types/read-only/configurator-attribute-read-only.component.html [hunk 18]
```diff
@@ -69,7 +74,7 @@
                 }
         }}
       </span>
-      <div class="cx-read-only-label">
+      <div class="cx-read-only-label" aria-hidden="true">
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/read-only/configurator-attribute-read-only.component.html:77]: Why was that needed for expert mode?

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts [hunk 19]
```diff
@@ -483,4 +533,32 @@ describe('ConfigurationFormComponent', () => {
...
   });
+
+  describe('expMode', () => {
+    it("should check whether expert mode status is set to 'true'", () => {
+      createComponent();
+      spyOn(configExpertModeService, 'getExpModeActive').and.returnValue(
+        of(true)
+      );
+
+      component.expMode
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts]: Please fix strict mode issue

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts [hunk 20]
```diff
@@ -483,4 +533,32 @@ describe('ConfigurationFormComponent', () => {
...
+        .unsubscribe();
+    });
+
+    it("should check whether expert mode status is set to 'false'", () => {
+      createComponent();
+      spyOn(configExpertModeService, 'getExpModeActive').and.returnValue(
+        of(false)
+      );
+
+      component.expMode
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts]: Please fix strict mode issue

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts [hunk 21]
```diff
@@ -393,6 +415,34 @@ describe('ConfigurationFormComponent', () => {
...
+          queryParams: { expMode: 'true' },
+        },
+      });
+      createComponent().ngOnInit();
+      expect(configExpertModeService.setExpModeRequested).toHaveBeenCalledTimes(
+        1
+      );
+    });
+
+    it('should not call setExpMode method', () => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts:434]: Please remove or update this test, because it is quite confusing. Maybe better to call without any query parameters at all.
Because it doesn't make a difference if line 438 is `queryParams: { expMode: 'false' } `or `queryParams: { expMode: 'truee' }` or `queryParams: { expMode: 'huhu' }`

The reason why expMode: 'false' resets the expert mode is only because a re-load of the page is triggered

File: feature-libs/product-configurator/rulebased/components/form/index.ts [hunk 22]
```diff
@@ -7,3 +7,4 @@
 export * from './configurator-form.component';
 export * from './configurator-form.event';
 export * from './configurator-form.module';
+export * from '../../core/events/configurator-logout-event.listener';
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/form/index.ts]: Remove this line

File: feature-libs/product-configurator/rulebased/core/index.ts [hunk 23]
```diff
@@ -8,4 +8,5 @@ export * from './config/index';
 export * from './connectors/index';
 export * from './facade/index';
 export * from './rulebased-configurator-core.module';
+export * from './services/index';
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/core/index.ts:11]: Also propagate index from events package!

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts [hunk 24]
```diff
@@ -29,33 +29,74 @@ import {
...
         this.occEndpointsService.buildUrl('createVariantConfiguration', {
           urlParams: { productCode },
+          queryParams: { expMode },
         })
       )
       .pipe(
         this.converterService.pipeable(VARIANT_CONFIGURATOR_NORMALIZER),
+        tap((resultConfiguration) => {
+          resultConfiguration.kbKey
+            ? this.setEpxModeActive(true)
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts]: Probably better: this.setEpxModeActive(resultConfiguration.kbKey !== undefined), would you agree?

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts [hunk 25]
```diff
@@ -29,33 +29,74 @@ import {
...
+  protected getExpModeRequested(): boolean {
+    let expMode = false;
+    this.configExpertModeService
+      ?.getExpModeRequested()
+      .pipe(take(1))
+      .subscribe((mode) => (expMode = mode));
+    return expMode;
+  }
+
+  protected setEpxModeActive(expMode: boolean) {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts]:
```suggestion
  protected setExpModeActive(expMode: boolean) {
```
Typo

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts [hunk 26]
```diff
@@ -70,15 +111,21 @@ export class VariantConfiguratorOccAdapter
...
         this.occEndpointsService.buildUrl('readVariantConfiguration', {
           urlParams: { configId },
-          queryParams: { groupId },
+          queryParams: { groupId, expMode },
         })
       )
       .pipe(
         this.converterService.pipeable(VARIANT_CONFIGURATOR_NORMALIZER),
+        tap((resultConfiguration) => {
+          resultConfiguration.kbKey
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts]: See above

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts [hunk 27]
```diff
@@ -107,6 +156,11 @@ export class VariantConfiguratorOccAdapter
       .patch<OccConfigurator.Configuration>(url, occConfiguration)
       .pipe(
         this.converterService.pipeable(VARIANT_CONFIGURATOR_NORMALIZER),
+        tap((resultConfiguration) => {
+          resultConfiguration.kbKey
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts]: See above

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/core/index.ts (hunk 23 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/core/index.ts:11]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts (hunk 25 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts (hunk 24 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts (hunk 27 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts (hunk 19 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts (hunk 20 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/form/index.ts (hunk 22 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/form/index.ts]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts (hunk 26 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts (hunk 21 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/form/configurator-form.component.spec.ts:434]: Removed!

File: feature-libs/product-configurator/rulebased/components/attribute/types/read-only/configurator-attribute-read-only.component.html (hunk 17 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/read-only/configurator-attribute-read-only.component.html:3]: Yes!

File: feature-libs/product-configurator/rulebased/components/attribute/types/read-only/configurator-attribute-read-only.component.html (hunk 18 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/read-only/configurator-attribute-read-only.component.html:77]: Yes!

**Review by ChristophHi — APPROVED** [review]
